tests/functional/testAuthenticationInApp.js

```diff
@@ -1027,4 +1027,885 @@ describe('User authentication', function() {
+            assert.strictEqual("impression", enterPasswordViewEventExtras.action);
+        });
+
+        it("back button event is recorded", async () => {
+            // Click the "<-" button
+            await vpn.waitForQueryAndClick(queries.screenAuthenticationInApp.AUTH_SIGNIN_BACK_BUTTON.visible());
+            await vpn.testLastInteractionEvent({
+                eventName: "authenticationInappStep",
+                elementId: "back_arrow",
+                screen
+            });
+        });
+
+        it("get help event is recorded", async () => {
+            // Click the "Get help" button
+            await vpn.waitForQueryAndClick(queries.screenAuthenticationInApp.AUTH_SIGNIN_GET_HELP_LINK.visible());
+            await vpn.testLastInteractionEvent({
+                eventName: "getHelp",
+                elementId: "get_help",
+                screen
+            });
+        });
+
+        it("paste button event is recorded", async () => {
+            // Click the "Paste" button
+            await vpn.waitForQueryAndClick(queries.screenAuthenticationInApp.AUTH_SIGNIN_PASSWORD_PASTE_BUTTON.visible());
+            await vpn.testLastInteractionEvent({
+                eventName: "authenticationInappStep",
+                elementId: "paste",
+                screen
+            });
+        });
+
+        it("change email button event is recorded", async () => {
+            // Click the "Change email" button
+            await vpn.waitForQueryAndClick(queries.screenAuthenticationInApp.AUTH_SIGNIN_CHANGE_EMAIL_BUTTON.visible());
+            await vpn.testLastInteractionEvent({
+                eventName: "authenticationInappStep",
+                elementId: "change_email",
+                screen
+            });
+        });
+
+        it("forgot password button event is recorded", async () => {
+            // Click the "Forgot your password?" button
+            await vpn.waitForQueryAndClick(queries.screenAuthenticationInApp.AUTH_SIGNIN_FORGOT_PASSWORD_BUTTON.visible());
+            await vpn.testLastInteractionEvent({
+                eventName: "authPasswordResetRequest",
+                elementId: "forgot_your_password",
+                screen
+            });
+        });
+
+        it.skip("cancel event is recorded", async () => {
+            // Click the "Cancel" button
+            await vpn.waitForQueryAndClick(queries.screenAuthenticationInApp.AUTH_SIGNIN_CANCEL_BUTTON.visible());
+            await vpn.testLastInteractionEvent({
+                eventName: "authenticationAborted",
+                elementId: "cancel",
+                screen
+            });
+        });
+
+        // Leaving this for documentation more than anything.
+        // The reason why it's skipped is because we can't query the "Terms of Service"
+        // or the "Privacy Policy" links, because they are not actual QML components.
+        // They are HTML tags inside of a QML text component???
```

Comment:
Calling on the QML experts to let me know if it is actually possible to query these elements cc @lesleyjanenorton @MattLichtenstein 

Essentially, what I'd need to here is query these `a` tags declared here https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/translations/strings.yaml#L689 with QML path.

Reply:
If I had the coordinates I could just use the click function though. The issue here is that I have no coordinates :/

Reply:
Would it make sense to have the functional test click on hard-coded coordinates, and if the link is not opened, just skip the telemetry verification? If I understand correctly, this test is strictly testing telemetry on link click, so we can do our best to tell the test where the link is

Reply:
I guess in that case I'd prefer to just not test it and leave it as is for documentation purposes. It would be too fragile to add hard coded coordinates. WDYT?